packages/e2e-tests/helpers/elements-panel.ts

```diff
@@ -293,7 +297,216 @@ export async function toggleMarkupNode(page: Page, locator: Locator, open: boole
+}
+
+// `doc_stacking.html` data and utils
+
+export interface StackingTestCase {
+  id: string;
+  position: { x: number; y: number };
+  expectedRules: string[];
+}
+
+// This recording  has 4x4 sets of carefully positioned DOM nodes on page
+// Many of these boxes overlap in some way, with various CSS position rules in effect.
+// The goal of this test is to verify that when we click at a specific coordinate,
+// the backend correctly determines the right mouse target underneath the cursor,
+// and we select the expected DOM node each time.
+// This means we have very specific expectations that if I click coords X,Y, we
+// will select a div with ".box1" or ".box2" or similar out of that group.
+
+export const stackingTestCases: StackingTestCase[] = [
+  // elements are stacked in the order in which they appear in the DOM (from back to front)
+  {
+    id: "r0c0",
+    position: { x: 40, y: 40 },
+    expectedRules: [".box2"],
+  },
+  // positioned elements are in front of unpositioned ones
+  {
+    id: "r0c1",
+    position: { x: 140, y: 40 },
+    expectedRules: [".box1"],
+  },
+  // parent elements are in front of children with a negative z-index
+  {
+    id: "r0c2",
+    position: { x: 240, y: 40 },
+    expectedRules: ["element", "body > div"],
+  },
+  // elements with a higher z-index are in front of those with a lower one
+  {
+    id: "r0c3",
+    position: { x: 340, y: 40 },
+    expectedRules: ["element", ".box1"],
+  },
+  // floating elements are in front of unpositioned elements
+  {
+    id: "r1c0",
+    position: { x: 40, y: 130 },
+    expectedRules: ["element", ".box1"],
+  },
+  // elements with visibility: hidden are ignored
+  {
+    id: "r1c1",
+    position: { x: 140, y: 140 },
+    expectedRules: [".box1"],
+  },
+  // elements with pointer-events: none are ignored
+  {
+    id: "r1c2",
+    position: { x: 240, y: 140 },
+    expectedRules: [".box1"],
+  },
+  // the z-index is ignored for unpositioned elements
+  {
+    id: "r1c3",
+    position: { x: 340, y: 140 },
+    expectedRules: [".box2"],
+  },
+
+  // the z-index is _not_ ignored for unpositioned elements if they are flex items
+  {
+    id: "r2c0",
+    position: { x: 40, y: 240 },
+    expectedRules: ["element", ".box1"],
+  },
+  // the z-index is _not_ ignored for unpositioned elements if they are grid items
+  {
+    id: "r2c1",
+    position: { x: 140, y: 240 },
+    expectedRules: ["element", ".box1"],
+  },
+  // positioned elements without z-index don't create a stacking context
+  {
+    id: "r2c2",
+    position: { x: 240, y: 240 },
+    expectedRules: ["element", ".box1"],
+  },
+  // children of positioned elements are in front of children of unpositioned ones
+  {
+    id: "r2c3",
+    position: { x: 240, y: 240 },
+    expectedRules: ["element", ".box1"],
+  },
+
+  // The last row has 4 sets of boxes, but some additional checks
+
+  // TODO [FE-626] This lower-left set is finicky and not working right with this  test setup atm.
```

Comment:
Is this comment still relevant? FE-626 is over a year old and seems pretty stale.

Looks like it was copied over verbatim from `packages/e2e-tests/tests/stacking.test.ts` – might be worth taking a quick look to see if it's worth copying this over?

Reply:
I can check to see if the behavior is any better.

Reply:
Gotcha. That's a bummer but if you think it's out of scope of this PR, that's understandable